### DISCUSSION TITLE
Does having "tracing" enabled cause a performance penality?

### DISCUSSION THREAD

**Discussion by IlCallo** [prisma/optimize-feedback#13] [General]
I just finished to try out the feature, and it's pretty interesting
I'd like to avoid removing and adding it every time I need to use it, but I'm afraid that leaving "tracing" enabled could cause some kind of performance penalty

Is this the case and should I remove it from my "prisma.schema" every time I don't need it anymore?

**Comment by millsp** [comment]
Hey, tracing is asynchronous and it will not slow down your queries. Of course, just like any code, tracing code is executed and will consume a little bit of CPU time but these aren't time consuming operations and the perf impact is really negligible. As for Optimize, it batches the data being sent in the background, it optimizes both on network and CPU as best as it can. These settings can be controlled. By the way, we have released Prisma Optimize into GA, you might want to check our new docs out https://www.prisma.io/data-platform/optimize.
